### ISSUE TITLE
loss nan

### ISSUE THREAD

**Issue by yqbeyond** [ronghuaiyang/arcface-pytorch#32]
I am training lenet5([Lenet5](https://github.com/activatedgeek/LeNet-5)) with arcface implemented here, but the loss changes to `nan` after several batches. Hope for help. Thanks.

![image](https://user-images.githubusercontent.com/16716680/61169118-a80d6c80-a58b-11e9-8744-a67608c6bf89.png)

**Comment by yqbeyond** [comment]
I have solved this by change `1` to `1.000001`.

![image](https://user-images.githubusercontent.com/16716680/61173078-c6da2600-a5c0-11e9-81dc-ab3a8e52a686.png)

**Comment by erow** [comment]
I met the same problem too. I solved this by following:
`sine = torch.sqrt(torch.clamp((1.0 - torch.pow(cosine, 2)),1e-9,1))`
